Python 3 compatibility

Hi, it seems that the Web UI works well with Python 3 and dulwich 0.13.0. However, Smart HTTP does not:
```
127.0.0.1 - - [26/Apr/2016 14:41:48] "GET /klaus/info/refs?service=git-upload-pack HTTP/1.1" 403 -
Error on request:
Traceback (most recent call last):
  File "/usr/lib64/python3.5/site-packages/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/usr/lib64/python3.5/site-packages/werkzeug/serving.py", line 184, in execute
    write(data)
  File "/usr/lib64/python3.5/site-packages/werkzeug/serving.py", line 164, in write
    assert isinstance(data, bytes), 'applications must write bytes'
AssertionError: applications must write bytes
```
I don't know if this is an issue with dulwich or klaus but I thought it would be nice to have an open bug to track the state of Python 3 compatibility.

--- a/klaus/highlighting.py
+++ b/klaus/highlighting.py
@@ -1,3 +1,5 @@
+from six.moves import filter
+
 from pygments import highlight
 from pygments.lexers import get_lexer_by_name, get_lexer_for_filename, \
                             guess_lexer, ClassNotFound, TextLexer
@@ -34,17 +36,18 @@ class KlausDefaultFormatter(HtmlFormatter):
 
     def _lookup_ctag(self, token):
         matches = list(self._get_all_ctags_matches(token))
-        best_matches = self.get_best_ctags_matches(matches)
+        best_matches = list(self.get_best_ctags_matches(matches))
         if not best_matches:
             return None, None
         else:
-            return best_matches[0]['file'], best_matches[0]['lineNumber']
+            return (best_matches[0]['file'].decode("utf-8"),
+                    best_matches[0]['lineNumber'])
 
     def _get_all_ctags_matches(self, token):
-        FIELDS = ('file', 'lineNumber', 'kind', 'language')
+        FIELDS = ('file', 'lineNumber', 'kind', b'language')
         from ctags import TagEntry
         entry = TagEntry()  # target "buffer" for ctags
-        if self._ctags.find(entry, token, 0):
+        if self._ctags.find(entry, token.encode("utf-8"), 0):
             yield dict((k, entry[k]) for k in FIELDS)
             while self._ctags.findNext(entry):
                 yield dict((k, entry[k]) for k in FIELDS)
@@ -53,7 +56,7 @@ class KlausDefaultFormatter(HtmlFormatter):
         if self.language is None:
             return matches
         else:
-            return filter(lambda match: match['language'] == self.language, matches)
+            return filter(lambda match: match[b'language'] == self.language.encode("utf-8"), matches)
 
 
 class KlausPythonFormatter(KlausDefaultFormatter):
@@ -66,7 +69,7 @@ class KlausPythonFormatter(KlausDefaultFormatter):
         # import of the tag in some other file.  We change the tag lookup mechanics
         # so that non-import matches are always preferred over import matches.
         return filter(
-            lambda match: match['kind'] != 'i',
+            lambda match: match['kind'] != b'i',
             super(KlausPythonFormatter, self).get_best_ctags_matches(matches)
         )
 

--- a/klaus/views.py
+++ b/klaus/views.py
@@ -207,7 +207,7 @@ class BaseFileView(TreeViewMixin, BaseBlobView):
                 self.context['commit'].id
             )
             ctags_args = {
-                'ctags': ctags.CTags(ctags_tagsfile),
+                'ctags': ctags.CTags(ctags_tagsfile.encode("utf-8")),
                 'ctags_baseurl': ctags_base_url,
             }
         else:

